sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/signaling/TrouterUtils.java

```diff
@@ -0,0 +1,343 @@
+package com.azure.android.communication.chat.signaling;
+
+import com.azure.android.communication.chat.signaling.properties.ChatEventId;
+import com.microsoft.trouterclient.ITrouterResponse;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+public class TrouterUtils {
+    static Map<ChatEventId, Integer> eventIdsMapping = new HashMap<ChatEventId, Integer>() {{
+        put(ChatEventId.chatMessageReceived, 200);
+        put(ChatEventId.typingIndicatorReceived, 245);
+        put(ChatEventId.readReceiptReceived, 246);
+        put(ChatEventId.chatMessageEdited, 247);
+        put(ChatEventId.chatMessageDeleted, 248);
+        put(ChatEventId.chatThreadCreated, 257);
+        put(ChatEventId.chatThreadPropertiesUpdated, 258);
+        put(ChatEventId.chatThreadDeleted, 259);
+        put(ChatEventId.participantsAdded, 260);
+        put(ChatEventId.participantsRemoved, 261);
+    }};
+
+    public static JSONObject toMessageHandler(String chatEventId, ITrouterResponse iTrouterResponse) {
+        int eventId = 0;
+        JSONObject genericPayload = null;
+        try {
+            eventId = eventIdsMapping.get(chatEventId);
+            genericPayload = new JSONObject(iTrouterResponse.getBody());
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+
+        try {
+            if (genericPayload == null || genericPayload.getInt("_eventId") != eventId) {
+                return null;
+            }
+        } catch (JSONException e) {
+            System.out.println(e.getMessage());
+        }
+
+        return toEventPayload(chatEventId, genericPayload);
+    }
+
+    public static JSONObject toEventPayload(String eventId, JSONObject payload) {
+        if (eventId.equals(ChatEventId.chatMessageReceived.name())) {
```

Comment:
Since the value of `eventId` does not change and this method will only enter one of the many if blocks below, wouldn't it make more sense to modularize this logic into smaller methods like `toChatMessageReceivedPayload` or `createChatMessageReceivedPayload` use a switch to call the right one based on the `eventId` contents?

Reply:
I just follow other languages, they put in one method instead of creating methods.

Reply:
I understand that but I think we should strive to make code as best as we can. I think it's better for readability and maintainability to avoid methods this big that do only one of many things based on a single condition. The fact that other languages do it does not necessarily mean that it's the best way to do things. It's not a blocker to approve this PR, but I think it's important to keep this in mind.